intellij-rust-native-helper/src/rustc_wrapper.rs

```diff
@@ -0,0 +1,29 @@
+use std::ffi::OsString;
+use std::process::{Command, Stdio};
+use std::io;
+
+pub struct ExitCode(pub Option<i32>);
+
+pub fn run_rustc_skipping_cargo_checking(
+    rustc_executable: OsString,
+    args: Vec<OsString>,
+) -> io::Result<ExitCode> {
+    let is_cargo_check = args.iter().any(|a| {
+        let a = a.to_string_lossy();
+        a.starts_with("--emit=") && a.contains("metadata") && !a.contains("link")
```

Comment:
Let's also add a comment that explains this condition and add this https://doc.rust-lang.org/rustc/command-line-arguments.html#--emit-specifies-the-types-of-output-files-to-generate url